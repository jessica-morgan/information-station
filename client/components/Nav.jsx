import React from 'reactn'
import { selectCategory } from '../utils'
import { Link } from 'react-router-dom'

const Nav = () => {
  return (
    <div style={{ display: 'block', width: '20vw', height: '100%', float: 'left', position: 'fixed' }}>
      <h2>News</h2>
      <Link to='/nzgFeed' onClick={() => selectCategory()}><h3>General</h3></Link>
      <Link to='/nzstFeed' onClick={() => selectCategory()}><h3>Science and Technology</h3></Link>
      <Link to='/jsFeed' onClick={() => selectCategory()}><h3>Javascript</h3></Link>
      <Link to='/qcFeed' onClick={() => selectCategory()}><h3>Quantum Computing</h3></Link>
      <Link to='/qpFeed' onClick={() => selectCategory()}><h3>Quantum Physics</h3></Link>
      <Link to='spaceFeed' onClick={() => selectCategory()}><h3>Space</h3></Link>
      <br/>
      <h2>Reddit</h2>
      <Link to='/phFeed' onClick={() => selectCategory()}><h3>Programmer Humor</h3></Link>
      <Link to='/rcFeed' onClick={() => selectCategory()}><h3>Retconned</h3></Link>
      <Link to='/bsFeed' onClick={() => selectCategory()}><h3>Is It Bullshit</h3></Link>
      <Link to='tataFeed' onClick={() => selectCategory()}><h3>Too Afraid To Ask</h3></Link>
      <br/>
      <Link to='/apod' onClick={() => selectCategory()}><h3>Astronomy Picture of the Day</h3></Link>
    </div>
  )
}

export default Nav
