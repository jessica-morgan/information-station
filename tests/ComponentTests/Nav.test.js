import React from 'reactn'
import Nav from '../../client/components/Nav'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme'

test('Nav renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Nav />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Nav contains this text', () => {
  const wrapper = shallow(<Nav />)
  expect(wrapper.text()).toMatch('New Zealand' && 'Reddit' && 'News')
})
