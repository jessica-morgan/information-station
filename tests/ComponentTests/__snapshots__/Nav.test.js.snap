// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<div
  style={
    Object {
      "display": "block",
      "float": "left",
      "height": "100%",
      "position": "fixed",
      "width": "20vw",
    }
  }
>
  <h2>
    New Zealand
  </h2>
  <a
    href="/nzgFeed"
    onClick={[Function]}
  >
    <h2>
      General
    </h2>
  </a>
  <a
    href="/nzstFeed"
    onClick={[Function]}
  >
    <h2>
      Science and Technology
    </h2>
  </a>
  <br />
  <h2>
    Reddit
  </h2>
  <a
    href="/phFeed"
    onClick={[Function]}
  >
    <h3>
      Programmer Humor
    </h3>
  </a>
  <a
    href="/rcFeed"
    onClick={[Function]}
  >
    <h3>
      Retconned
    </h3>
  </a>
  <a
    href="/bsFeed"
    onClick={[Function]}
  >
    <h3>
      Is It Bullshit
    </h3>
  </a>
  <a
    href="/tataFeed"
    onClick={[Function]}
  >
    <h3>
      Too Afraid To Ask
    </h3>
  </a>
  <br />
  <h2>
    News
  </h2>
  <a
    href="/jsFeed"
    onClick={[Function]}
  >
    <h3>
      Javascript
    </h3>
  </a>
  <a
    href="/qcFeed"
    onClick={[Function]}
  >
    <h3>
      Quantum Computing
    </h3>
  </a>
  <a
    href="/qpFeed"
    onClick={[Function]}
  >
    <h3>
      Quantum Physics
    </h3>
  </a>
  <a
    href="/spaceFeed"
    onClick={[Function]}
  >
    <h3>
      Space
    </h3>
  </a>
  <br />
  <a
    href="/apod"
    onClick={[Function]}
  >
    <h2>
      Astronomy Picture of the Day
    </h2>
  </a>
</div>
`;
